### DISCUSSION TITLE
QueuedInterceptor token updating problem

### DISCUSSION THREAD

**Discussion by I-m-good-man** [cfug/dio#2302] [Q&A] [answered]
I understand correctly that the QueuedInterceptor works as follows?:
The interceptor in the queue sends a request and blocks the processing of further requests until it receives a response to the sent request and processes it in the response/error processing method.

If it doesn't work that way, then there is the following problem with updating tokens in the interseptor's response processing method:
If the client sends 2 requests to the server with an invalid bearer token, it receives 2 responses with the error 401 Unauthorized (the token is not valid). The QueuedInterceptor must process these 2 erroneous responses in turn - in each call to the processing method, the interсeptor updates a pair of tokens through a refresh token. 
After processing the first response, a pair of tokens was updated and now the bearer token on the client side is valid. But, the interceptor must also process the second 401 response. The interceptor does not know that a pair of tokens has already been updated, so it initiates a re-update of an already fresh pair of tokens. 
And there can be more than two requests with an invalid token, so the QueuedInterceptor can initiate many updates of a pair of tokens.

**Comment by AlexV525** [comment, marked as answer]
https://github.com/cfug/dio/blob/main/example_dart/lib/queued_interceptor_crsftoken.dart

**Reply by scopendo** [reply]
I think there's a minor bug in this example on line 65 `if (requestedAccessToken == tokenManager.accessToken) {`

`requestedAccessToken` will contain the bearer prefix, e.g. `Bearer {old_access_token}`.

**Reply by kuhnroyal** [reply]
Yea I think you are right. Would you mind opening a PR with a fix?
